dLux/apertures.py

```diff
@@ -0,0 +1,2641 @@
+        
+        Parameters:
+        -----------
+        x1: float, meters
+            The x coordinates of the points that are to be connected. 
+        y1: float, meters
+            The y coordinates of the points that are to be connected. 
+            Must have the same shape as x. 
+            
+        Returns:
+        --------
+        ms: float, None (meters / meter)
+            The gradients of the lines that connect the vertices. The 
+            vertices wrap around to form a closed shape whatever it 
+            may look like. 
+        """
+        x_diffs: float = x1 - np.roll(x1, -1)
+        y_diffs: float = y1 - np.roll(y1, -1)
+        return y_diffs / x_diffs
+    
+    
+    def _extent(self: ApertureLayer) -> float:
+        """
+        Returns the largest distance to the outer edge of the aperture from the
+        centre. For inherited classes, consider implementing analytically for speed.
+
+        Parameters
+        ----------
+        coordinates : Array
+            The cartesian coordinates to generate the hexikes on.
+            The dimensions of the tensor should be `(2, npix, npix)`.
+            where the leading axis is the x and y dimensions.  
+
+        Returns
+        -------
+        extent : float
+            The maximum distance from centre to edge of aperture
+        """
+        verts: float = self.vertices
+        dist_to_verts: float = np.hypot(verts[:, 1], verts[:, 0])
+        return np.max(dist_to_verts)
+    
+    
+    def _metric(self: ApertureLayer, coords: float) -> float:
+        """
+        A measure of how far a pixel is from the aperture.
+        This is a very abstract description that was constructed 
+        when dealing with the soft edging. For a normal binary 
+        representation the metric is zero if it is inside the
+        aperture and one if it is outside the aperture. Notice,
+        we have not attempted to prove that this is a metric 
+        via the axioms, this is just a handy name that brings 
+        to mind the general idea. For a soft edged aperture the 
+        metric is different.
+
+        Parameters:
+        -----------
+        distances: Array
+            The distances of each pixel from the edge of the aperture. 
+            Again, the words distances is designed to aid in 
+            conveying the idea and is not strictly true. We are
+            permitting negative distances when inside the aperture
+            because this was simplest to implement. 
+
+        Returns:
+        --------
+        non_occ_ap: Array 
+            This is essential the final step in processing to produce
+            the aperture. What is returned is the non-occulting 
+            version of the aperture. 
+        """
+        bc_x1: float = self.vertices[:, 0][:, None, None]
+        bc_y1: float = self.vertices[:, 1][:, None, None]
+
+        bc_x: float = coords[0][None, :, :]
+        bc_y: float = coords[1][None, :, :]
```

Comment:
I would just add some one-liners describing what `bc_x1` are etc. It's a little hard to see _what_ this function is doing with these somewhat abstract names